I have changed everything but I am not able to commit.
I am getting such error:
```
hassfest.................................................................Failed
- hook id: hassfest
- exit code: 1

Error loading libturbojpeg; Camera snapshot performance will be sub-optimal
Traceback (most recent call last):
  File "/workspaces/core/homeassistant/components/camera/img_util.py", line 100, in __init__
    TurboJPEGSingleton.__instance = TurboJPEG()
                                    ^^^^^^^^^
NameError: name 'TurboJPEG' is not defined
Warning: homeassistant/components/dte_energy_bridge/__init__.py and manifest.json missing, skipping directory. If this is your development environment, you can safely delete this folder.
Validating application_credentials... done in 0.09s
Validating bluetooth... done in 0.01s
Validating codeowners... done in 0.01s
Validating config_schema... done in 0.65s
Validating dependencies... done in 1.81s
Validating dhcp... done in 0.01s
Validating icons... done in 0.21s
Validating json... done in 0.00s
Validating manifest... done in 0.02s
Validating mqtt... done in 0.01s
Validating quality_scale... done in 0.12s
Validating requirements... done in 0.01s
Validating services... done in 0.61s
Validating ssdp... done in 0.01s
Validating translations... done in 3.49s
Validating usb... done in 0.01s
Validating zeroconf... done in 0.01s
Validating config_flow... done in 0.06s
Validating docker... done in 0.00s
Validating mypy_config... done in 0.04s
Validating metadata... done in 0.00s

Integrations: 1304
Invalid integrations: 1

Found errors. Generating files canceled.

Integration assist_satellite:
* [ERROR] [SERVICES] Invalid services.yaml: Unknown supported feature 'assist_satellite.AssistSatelliteEntityFeature.ANNOUNCE' @ data['announce']['target']['entity'][0]['supported_features'][0]. Got None
```
I've rebased to upstream/dev, Dev container is recreated and still there are some issues.

--- a/homeassistant/components/watergate/__init__.py
+++ b/homeassistant/components/watergate/__init__.py
@@ -10,14 +10,18 @@ from watergate_local_api import WatergateLocalApiClient
 from watergate_local_api.models import WebhookEvent
 
 from homeassistant.components.http import HomeAssistantView
-from homeassistant.components.webhook import Request, Response, async_generate_url
+from homeassistant.components.webhook import (
+    Request,
+    Response,
+    async_generate_url,
+    async_register,
+)
 from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import CONF_IP_ADDRESS, CONF_NAME, CONF_WEBHOOK_ID, Platform
+from homeassistant.const import CONF_IP_ADDRESS, CONF_WEBHOOK_ID, Platform
 from homeassistant.core import HomeAssistant
 
 from .const import DOMAIN
 from .coordinator import WatergateDataCoordinator
-from .entity import WatergateConfigEntry, WatergateData
 
 _LOGGER = logging.getLogger(__name__)
 
@@ -26,26 +30,28 @@ PLATFORMS: list[Platform] = [
     Platform.VALVE,
 ]
 
+type WatergateConfigEntry = ConfigEntry[WatergateDataCoordinator]
+
 
 async def async_setup_entry(hass: HomeAssistant, entry: WatergateConfigEntry) -> bool:
     """Set up Watergate from a config entry."""
-    sonic_name = entry.data[CONF_NAME]
     sonic_address = entry.data[CONF_IP_ADDRESS]
     webhook_id = entry.data[CONF_WEBHOOK_ID]
 
     _LOGGER.debug(
-        "Setting up watergate local api integration for device: %s (IP: %s)",
-        sonic_name,
+        "Setting up watergate local api integration for device: IP: %s)",
         sonic_address,
     )
 
-    watergate_client = WatergateLocalApiClient(sonic_address)
+    watergate_client = WatergateLocalApiClient(
+        sonic_address if sonic_address.startswith("http") else f"http://{sonic_address}"
+    )
 
     coordinator = WatergateDataCoordinator(hass, watergate_client)
+    entry.runtime_data = coordinator
 
-    hass.components.webhook.async_unregister(webhook_id)
-    hass.components.webhook.async_register(
-        DOMAIN, "Watergate", webhook_id, get_webhook_handler(coordinator)
+    async_register(
+        hass, DOMAIN, "Watergate", webhook_id, get_webhook_handler(coordinator)
     )
 
     _LOGGER.debug("Registered webhook: %s", webhook_id)
@@ -53,18 +59,18 @@ async def async_setup_entry(hass: HomeAssistant, entry: WatergateConfigEntry) ->
     await coordinator.async_config_entry_first_refresh()
 
     await watergate_client.async_set_webhook_url(
-        async_generate_url(hass, webhook_id, allow_ip=True)
+        async_generate_url(hass, webhook_id, allow_ip=True, prefer_external=False)
     )
-
-    entry.runtime_data = WatergateData(watergate_client, coordinator, sonic_name)
 
     await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
 
     return True
 
 
-async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+async def async_unload_entry(hass: HomeAssistant, entry: WatergateConfigEntry) -> bool:
     """Unload a config entry."""
+    webhook_id = entry.data[CONF_WEBHOOK_ID]
+    hass.components.webhook.async_unregister(webhook_id)
     return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
 
 
@@ -91,8 +97,8 @@ def get_webhook_handler(
         body_type = body.get("type")
 
         coordinator_data = coordinator.data
-        if body_type == Platform.VALVE and coordinator_data.state:
-            coordinator_data.state.valve_state = data.state
+        if body_type == Platform.VALVE and coordinator_data:
+            coordinator_data.valve_state = data.state
 
         coordinator.async_set_updated_data(coordinator_data)
 

--- a/homeassistant/components/watergate/config_flow.py
+++ b/homeassistant/components/watergate/config_flow.py
@@ -1,17 +1,24 @@
 """Config flow for Watergate."""
 
+import logging
+
 import voluptuous as vol
-from watergate_local_api.watergate_api import WatergateLocalApiClient
+from watergate_local_api.watergate_api import (
+    WatergateApiException,
+    WatergateLocalApiClient,
+)
 
 from homeassistant.components.webhook import async_generate_id as webhook_generate_id
 from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
-from homeassistant.const import CONF_IP_ADDRESS, CONF_NAME, CONF_WEBHOOK_ID
+from homeassistant.const import CONF_IP_ADDRESS, CONF_WEBHOOK_ID
 
 from .const import DOMAIN
 
+_LOGGER = logging.getLogger(__name__)
+
+SONIC = "Sonic"
 WATERGATE_SCHEMA = vol.Schema(
     {
-        vol.Required(CONF_NAME): str,
         vol.Required(CONF_IP_ADDRESS): str,
     }
 )
@@ -26,24 +33,30 @@ class WatergateConfigFlow(ConfigFlow, domain=DOMAIN):
         """Handle a flow initiated by the user."""
         errors: dict[str, str] = {}
         if user_input is not None:
-            if not await self.validate_connection(user_input[CONF_IP_ADDRESS]):
+            watergate_client = WatergateLocalApiClient(
+                self.prepare_ip_address(user_input[CONF_IP_ADDRESS])
+            )
+            try:
+                state = await watergate_client.async_get_device_state()
+            except WatergateApiException as exception:
+                _LOGGER.error("Error connecting to Watergate device: %s", exception)
                 errors[CONF_IP_ADDRESS] = "cannot_connect"
             else:
-                return self.async_create_entry(
-                    title=user_input[CONF_NAME],
-                    data={**user_input, CONF_WEBHOOK_ID: webhook_generate_id()},
-                )
+                if state is None:
+                    _LOGGER.error("Device state returned as None")
+                    errors[CONF_IP_ADDRESS] = "cannot_connect"
+                else:
+                    await self.async_set_unique_id(state.serial_number)
+                    self._abort_if_unique_id_configured()
+                    return self.async_create_entry(
+                        data={**user_input, CONF_WEBHOOK_ID: webhook_generate_id()},
+                        title=SONIC,
+                    )
 
         return self.async_show_form(
             step_id="user", data_schema=WATERGATE_SCHEMA, errors=errors
         )
 
-    async def validate_connection(self, ip_address: str):
-        """Validate the connection to the Watergate device."""
-        try:
-            return (
-                await WatergateLocalApiClient(ip_address).async_get_device_state()
-                is not None
-            )
-        except Exception:  # noqa: BLE001
-            return False
+    def prepare_ip_address(self, ip_address: str) -> str:
+        """Prepare the IP address for the Watergate device."""
+        return ip_address if ip_address.startswith("http") else f"http://{ip_address}"

--- a/homeassistant/components/watergate/coordinator.py
+++ b/homeassistant/components/watergate/coordinator.py
@@ -14,15 +14,7 @@ from .const import DOMAIN
 _LOGGER = logging.getLogger(__name__)
 
 
-class WatergateAgregatedRequests:
-    """Class to hold aggregated requests."""
-
-    def __init__(self, state: DeviceState | None = None) -> None:
-        """Initialize aggregated requests."""
-        self.state = state
-
-
-class WatergateDataCoordinator(DataUpdateCoordinator[WatergateAgregatedRequests]):
+class WatergateDataCoordinator(DataUpdateCoordinator[DeviceState]):
     """Class to manage fetching watergate data."""
 
     def __init__(self, hass: HomeAssistant, api: WatergateLocalApiClient) -> None:
@@ -35,9 +27,9 @@ class WatergateDataCoordinator(DataUpdateCoordinator[WatergateAgregatedRequests]
         )
         self.api = api
 
-    async def _async_update_data(self) -> WatergateAgregatedRequests:
+    async def _async_update_data(self) -> DeviceState:
         try:
             state = await self.api.async_get_device_state()
         except WatergateApiException as exc:
             raise UpdateFailed from exc
-        return WatergateAgregatedRequests(state=state)
+        return state

--- a/homeassistant/components/watergate/entity.py
+++ b/homeassistant/components/watergate/entity.py
@@ -1,50 +1,30 @@
 """Watergate Base Entity Definition."""
 
-from dataclasses import dataclass
-
-from watergate_local_api import WatergateLocalApiClient
-
-from homeassistant.config_entries import ConfigEntry
-from homeassistant.const import CONF_NAME
 from homeassistant.helpers.device_registry import DeviceInfo
 from homeassistant.helpers.update_coordinator import CoordinatorEntity
 
 from .const import DOMAIN, MANUFACTURER
 from .coordinator import WatergateDataCoordinator
 
-type WatergateConfigEntry = ConfigEntry[WatergateData]
-
 
 class WatergateEntity(CoordinatorEntity[WatergateDataCoordinator]):
     """Define a base Watergate entity."""
 
+    _attr_has_entity_name = True
+
     def __init__(
         self,
         coordinator: WatergateDataCoordinator,
-        entry: WatergateConfigEntry,
         entity_name: str,
     ) -> None:
         """Initialize the entity."""
         super().__init__(coordinator)
-        self._attr_has_entity_name = True
         self._api_client = coordinator.api
-        self._attr_unique_id = f"{entry.data[CONF_NAME]}.{entity_name}"
-        self._entry = entry
+        self._attr_unique_id = f"{coordinator.data.serial_number}.{entity_name}"
         self._attr_device_info = DeviceInfo(
-            identifiers={(DOMAIN, entry.entry_id)},
-            name=entry.data[CONF_NAME],
+            identifiers={(DOMAIN, coordinator.data.serial_number)},
+            name="Sonic",
+            serial_number=coordinator.data.serial_number,
             manufacturer=MANUFACTURER,
-            sw_version=coordinator.data.state.firmware_version
-            if coordinator.data.state
-            else "",
+            sw_version=coordinator.data.firmware_version if coordinator.data else None,
         )
-        self._api_client = entry.runtime_data.client
-
-
-@dataclass
-class WatergateData:
-    """Data for the A. O. Smith integration."""
-
-    client: WatergateLocalApiClient
-    coordinator: WatergateDataCoordinator
-    sonic_name: str

--- a/homeassistant/components/watergate/manifest.json
+++ b/homeassistant/components/watergate/manifest.json
@@ -6,5 +6,6 @@
   "dependencies": ["http", "webhook"],
   "documentation": "https://www.home-assistant.io/integrations/watergate",
   "iot_class": "local_push",
-  "requirements": ["watergate-local-api==2024.1.11"]
+  "quality_scale": "bronze",
+  "requirements": ["watergate-local-api==2024.4.1"]
 }

--- a/homeassistant/components/watergate/quality_scale.yaml
+++ b/homeassistant/components/watergate/quality_scale.yaml
@@ -1,0 +1,43 @@
+rules:
+  # Bronze
+  action-setup:
+    status: exempt
+    comment: |
+      This integration does not provide additional actions.
+  appropriate-polling: done
+  brands: done
+  common-modules: done
+  config-flow-test-coverage: done
+  config-flow: done
+  dependency-transparency: done
+  docs-actions:
+    status: exempt
+    comment: |
+      This integration does not provide additional actions.
+  docs-high-level-description: done
+  docs-installation-instructions: done
+  docs-removal-instructions: done
+  entity-event-setup:
+    status: exempt
+    comment: |
+      Entities of this integration does not explicitly subscribe to events.
+  entity-unique-id: done
+  has-entity-name: done
+  runtime-data: done
+  test-before-configure: done
+  test-before-setup: done
+  unique-config-entry: done
+  # Silver
+  config-entry-unloading: done
+  log-when-unavailable: todo
+  entity-unavailable: done
+  action-exceptions: done
+  reauthentication-flow:
+    status: exempt
+    comment: |
+      This integration does not require authentication.
+  parallel-updates: done
+  test-coverage: done
+  integration-owner: done
+  docs-installation-parameters: todo
+  docs-configuration-parameters: todo

--- a/homeassistant/components/watergate/strings.json
+++ b/homeassistant/components/watergate/strings.json
@@ -1,28 +1,21 @@
 {
   "config": {
     "step": {
-      "confirm": {
-        "description": "[%key:common::config_flow::description::confirm_setup%]"
-      },
       "user": {
         "data": {
-          "name": "[%key:common::config_flow::data::device%]",
           "ip_address": "[%key:common::config_flow::data::ip%]"
         },
         "title": "Configure Watergate device",
-        "description": "Please put name and IP address of your Watergate device"
-      },
-      "reconfigure": {
-        "data": {
-          "name": "[%key:common::config_flow::data::device%]",
-          "ip_address": "[%key:common::config_flow::data::ip%]"
-        },
-        "title": "Reconfigure Watergate device",
-        "description": "[%key:component::watergate::config::step::user::description%]"
+        "data_description": {
+          "ip_address": "Provide an IP address of your Watergate device."
+        }
       }
     },
     "error": {
       "cannot_connect": "[%key:common::config_flow::error::cannot_connect%]"
+    },
+    "abort": {
+      "already_configured": "[%key:common::config_flow::abort::already_configured_device%]"
     }
   }
 }

--- a/homeassistant/components/watergate/valve.py
+++ b/homeassistant/components/watergate/valve.py
@@ -10,10 +10,12 @@ from homeassistant.components.valve import (
 from homeassistant.core import callback
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
+from . import WatergateConfigEntry
 from .coordinator import WatergateDataCoordinator
-from .entity import WatergateConfigEntry, WatergateEntity
+from .entity import WatergateEntity
 
 ENTITY_NAME = "valve"
+PARALLEL_UPDATES = 0
 
 
 async def async_setup_entry(
@@ -21,11 +23,9 @@ async def async_setup_entry(
     config_entry: WatergateConfigEntry,
     async_add_entities: AddEntitiesCallback,
 ) -> None:
-    """Set up all entries for Wolf Platform."""
+    """Set up all entries for Watergate Platform."""
 
-    async_add_entities(
-        [SonicValve(config_entry.runtime_data.coordinator, config_entry)]
-    )
+    async_add_entities([SonicValve(config_entry.runtime_data)])
 
 
 class SonicValve(WatergateEntity, ValveEntity):
@@ -40,13 +40,10 @@ class SonicValve(WatergateEntity, ValveEntity):
     def __init__(
         self,
         coordinator: WatergateDataCoordinator,
-        entry: WatergateConfigEntry,
     ) -> None:
         """Initialize the sensor."""
-        super().__init__(coordinator, entry, ENTITY_NAME)
-        self._valve_state = (
-            coordinator.data.state.valve_state if coordinator.data.state else None
-        )
+        super().__init__(coordinator, ENTITY_NAME)
+        self._valve_state = coordinator.data.valve_state if coordinator.data else None
 
     @property
     def is_closed(self) -> bool:
@@ -66,11 +63,9 @@ class SonicValve(WatergateEntity, ValveEntity):
     @callback
     def _handle_coordinator_update(self) -> None:
         """Handle data update."""
-        self._attr_available = self.coordinator.data.state is not None
+        self._attr_available = self.coordinator.data is not None
         self._valve_state = (
-            self.coordinator.data.state.valve_state
-            if self.coordinator.data.state
-            else None
+            self.coordinator.data.valve_state if self.coordinator.data else None
         )
         self.async_write_ha_state()
 
@@ -81,7 +76,7 @@ class SonicValve(WatergateEntity, ValveEntity):
         self.async_write_ha_state()
 
     async def async_close_valve(self, **kwargs: Any) -> None:
-        """Open the valve."""
-        await self._api_client.async_set_valve_state("close")
+        """Close the valve."""
+        await self._api_client.async_set_valve_state(ValveState.CLOSED)
         self._valve_state = ValveState.CLOSING
         self.async_write_ha_state()

--- a/requirements_all.txt
+++ b/requirements_all.txt
@@ -2981,7 +2981,7 @@ watchdog==2.3.1
 waterfurnace==1.1.0
 
 # homeassistant.components.watergate
-watergate-local-api==2024.1.11
+watergate-local-api==2024.4.1
 
 # homeassistant.components.weatherflow_cloud
 weatherflow4py==1.0.6

--- a/requirements_test_all.txt
+++ b/requirements_test_all.txt
@@ -2382,7 +2382,7 @@ wallbox==0.7.0
 watchdog==2.3.1
 
 # homeassistant.components.watergate
-watergate-local-api==2024.1.11
+watergate-local-api==2024.4.1
 
 # homeassistant.components.weatherflow_cloud
 weatherflow4py==1.0.6

--- a/tests/components/watergate/__init__.py
+++ b/tests/components/watergate/__init__.py
@@ -1,19 +1,6 @@
 """Tests for the Watergate integration."""
 
-from watergate_local_api.models import DeviceState
-
 from homeassistant.core import HomeAssistant
-
-DEFAULT_DEVICE_STATE = DeviceState(
-    "open",
-    "on",
-    True,
-    True,
-    "battery",
-    "1.0.0",
-    100,
-    {"volume": 1.2, "duration": 100},
-)
 
 
 async def init_integration(hass: HomeAssistant, mock_entry) -> None:

--- a/tests/components/watergate/conftest.py
+++ b/tests/components/watergate/conftest.py
@@ -5,10 +5,14 @@ from collections.abc import Generator
 import pytest
 
 from homeassistant.components.watergate.const import DOMAIN
-from homeassistant.const import CONF_IP_ADDRESS, CONF_NAME
+from homeassistant.const import CONF_IP_ADDRESS
 
-from . import DEFAULT_DEVICE_STATE
-from .const import MOCK_CONFIG, MOCK_WEBHOOK_ID
+from .const import (
+    DEFAULT_DEVICE_STATE,
+    DEFAULT_SERIAL_NUMBER,
+    MOCK_CONFIG,
+    MOCK_WEBHOOK_ID,
+)
 
 from tests.common import AsyncMock, MockConfigEntry, patch
 
@@ -18,7 +22,8 @@ def mock_watergate_client() -> Generator[AsyncMock]:
     """Fixture to mock WatergateLocalApiClient."""
     with (
         patch(
-            "homeassistant.components.watergate.WatergateLocalApiClient"
+            "homeassistant.components.watergate.WatergateLocalApiClient",
+            autospec=True,
         ) as mock_client_main,
         patch(
             "homeassistant.components.watergate.config_flow.WatergateLocalApiClient",
@@ -27,8 +32,6 @@ def mock_watergate_client() -> Generator[AsyncMock]:
     ):
         mock_client_instance = mock_client_main.return_value
 
-        mock_client_instance.async_set_webhook_url = AsyncMock(return_value=None)
-        mock_client_instance.async_set_valve_state = AsyncMock(return_value=None)
         mock_client_instance.async_get_device_state = AsyncMock(
             return_value=DEFAULT_DEVICE_STATE
         )
@@ -45,7 +48,7 @@ def mock_setup_entry() -> Generator[AsyncMock]:
 
 
 @pytest.fixture
-def mock_webhook_id_generatio() -> Generator[None]:
+def mock_webhook_id_generation() -> Generator[None]:
     """Fixture to mock webhook_id generation."""
     with patch(
         "homeassistant.components.watergate.config_flow.webhook_generate_id",
@@ -58,12 +61,11 @@ def mock_webhook_id_generatio() -> Generator[None]:
 def mock_entry() -> MockConfigEntry:
     """Create full mocked entry to be used in config_flow tests."""
     return MockConfigEntry(
-        version=1,
         domain=DOMAIN,
-        title="Existing Device",
+        title="Sonic",
         data=MOCK_CONFIG,
-        source="user",
         entry_id="12345",
+        unique_id=DEFAULT_SERIAL_NUMBER,
     )
 
 
@@ -71,6 +73,5 @@ def mock_entry() -> MockConfigEntry:
 def user_input() -> dict[str, str]:
     """Create user input for config_flow tests."""
     return {
-        CONF_NAME: "Test Device",
         CONF_IP_ADDRESS: "192.168.1.100",
     }

--- a/tests/components/watergate/const.py
+++ b/tests/components/watergate/const.py
@@ -1,5 +1,7 @@
 """Constants for the Watergate tests."""
 
+from watergate_local_api.models import DeviceState
+
 from homeassistant.const import CONF_IP_ADDRESS, CONF_NAME, CONF_WEBHOOK_ID
 
 MOCK_WEBHOOK_ID = "webhook_id"
@@ -9,3 +11,17 @@ MOCK_CONFIG = {
     CONF_IP_ADDRESS: "http://localhost",
     CONF_WEBHOOK_ID: MOCK_WEBHOOK_ID,
 }
+
+DEFAULT_SERIAL_NUMBER = "a63182948ce2896a"
+
+DEFAULT_DEVICE_STATE = DeviceState(
+    "open",
+    "on",
+    True,
+    True,
+    "battery",
+    "1.0.0",
+    100,
+    {"volume": 1.2, "duration": 100},
+    DEFAULT_SERIAL_NUMBER,
+)

--- a/tests/components/watergate/snapshots/test_valve.ambr
+++ b/tests/components/watergate/snapshots/test_valve.ambr
@@ -1,5 +1,5 @@
 # serializer version: 1
-# name: test_change_valve_state
+# name: test_change_valve_state_snapshot
   StateSnapshot({
     'attributes': ReadOnlyDict({
       'device_class': 'water',

--- a/tests/components/watergate/test_config_flow.py
+++ b/tests/components/watergate/test_config_flow.py
@@ -3,23 +3,22 @@
 from collections.abc import Generator
 
 import pytest
+from watergate_local_api import WatergateApiException
 
 from homeassistant.components.watergate.const import DOMAIN
 from homeassistant.config_entries import SOURCE_USER
-from homeassistant.const import CONF_IP_ADDRESS, CONF_NAME, CONF_WEBHOOK_ID
+from homeassistant.const import CONF_IP_ADDRESS, CONF_WEBHOOK_ID
 from homeassistant.data_entry_flow import FlowResultType
 
-from . import DEFAULT_DEVICE_STATE
-from .const import MOCK_WEBHOOK_ID
+from .const import DEFAULT_DEVICE_STATE, DEFAULT_SERIAL_NUMBER, MOCK_WEBHOOK_ID
 
-from tests.common import AsyncMock, HomeAssistant
+from tests.common import AsyncMock, HomeAssistant, MockConfigEntry
 
 
 async def test_step_user_form(
     hass: HomeAssistant,
     mock_watergate_client: Generator[AsyncMock],
-    mock_webhook_id_generatio: Generator[None],
-    mock_setup_entry: Generator[AsyncMock],
+    mock_webhook_id_generation: Generator[None],
     user_input: dict[str, str],
 ) -> None:
     """Test checking if registration form works end to end."""
@@ -30,7 +29,6 @@ async def test_step_user_form(
 
     assert result["type"] is FlowResultType.FORM
     assert result["step_id"] == "user"
-    assert CONF_NAME in result["data_schema"].schema
     assert CONF_IP_ADDRESS in result["data_schema"].schema
 
     result = await hass.config_entries.flow.async_configure(
@@ -38,21 +36,21 @@ async def test_step_user_form(
     )
 
     assert result["type"] is FlowResultType.CREATE_ENTRY
-    assert result["title"] == "Test Device"
+    assert result["title"] == "Sonic"
     assert result["data"] == {**user_input, CONF_WEBHOOK_ID: MOCK_WEBHOOK_ID}
+    assert result["result"].unique_id == DEFAULT_SERIAL_NUMBER
 
 
 @pytest.mark.parametrize(
     "client_result",
-    [AsyncMock(return_value=None), AsyncMock(side_effect=Exception)],
+    [AsyncMock(return_value=None), AsyncMock(side_effect=WatergateApiException)],
 )
 async def test_step_user_form_with_exception(
     hass: HomeAssistant,
     mock_watergate_client: Generator[AsyncMock],
     user_input: dict[str, str],
     client_result: AsyncMock,
-    mock_webhook_id_generatio: Generator[None],
-    mock_setup_entry: Generator[AsyncMock],
+    mock_webhook_id_generation: Generator[None],
 ) -> None:
     """Test checking if errors will be displayed when Exception is thrown while checking device state."""
     mock_watergate_client.async_get_device_state = client_result
@@ -80,5 +78,30 @@ async def test_step_user_form_with_exception(
     )
 
     assert result["type"] is FlowResultType.CREATE_ENTRY
-    assert result["title"] == "Test Device"
+    assert result["title"] == "Sonic"
     assert result["data"] == {**user_input, CONF_WEBHOOK_ID: MOCK_WEBHOOK_ID}
+
+
+async def test_abort_if_id_is_not_unique(
+    hass: HomeAssistant,
+    mock_watergate_client: Generator[AsyncMock],
+    mock_entry: MockConfigEntry,
+    user_input: dict[str, str],
+) -> None:
+    """Test checking if we will inform user that this entity is already registered."""
+    mock_entry.add_to_hass(hass)
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": SOURCE_USER}
+    )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "user"
+    assert CONF_IP_ADDRESS in result["data_schema"].schema
+
+    result = await hass.config_entries.flow.async_configure(
+        result["flow_id"], user_input
+    )
+
+    assert result["type"] is FlowResultType.ABORT
+    assert result["reason"] == "already_configured"

--- a/tests/components/watergate/test_init.py
+++ b/tests/components/watergate/test_init.py
@@ -24,7 +24,7 @@ async def test_async_setup_entry(
     hass.config.internal_url = "http://hassio.local"
 
     with (
-        patch("homeassistant.components.webhook.async_register") as mock_webhook,
+        patch("homeassistant.components.watergate.async_register") as mock_webhook,
     ):
         await init_integration(hass, mock_entry)
 

--- a/tests/components/watergate/test_valve.py
+++ b/tests/components/watergate/test_valve.py
@@ -13,7 +13,7 @@ from . import init_integration
 from tests.common import AsyncMock, MockConfigEntry
 
 
-async def test_change_valve_state(
+async def test_change_valve_state_snapshot(
     hass: HomeAssistant,
     snapshot: SnapshotAssertion,
     mock_watergate_client: Generator[AsyncMock],
@@ -29,6 +29,21 @@ async def test_change_valve_state(
     assert registered_entity.state == ValveState.OPEN
     assert registered_entity == snapshot
 
+
+async def test_change_valve_state(
+    hass: HomeAssistant,
+    mock_watergate_client: Generator[AsyncMock],
+    mock_entry: MockConfigEntry,
+) -> None:
+    """Test entities become unavailable after failed update."""
+    await init_integration(hass, mock_entry)
+
+    entity_id = "valve.sonic"
+
+    registered_entity = hass.states.get(entity_id)
+    assert registered_entity
+    assert registered_entity.state == ValveState.OPEN
+
     await hass.services.async_call(
         VALVE_DOMAIN,
         SERVICE_CLOSE_VALVE,
@@ -40,7 +55,7 @@ async def test_change_valve_state(
     assert registered_entity
     assert registered_entity.state == ValveState.CLOSING
 
-    mock_watergate_client.async_set_valve_state.assert_called_once_with("close")
+    mock_watergate_client.async_set_valve_state.assert_called_once_with("closed")
     mock_watergate_client.async_set_valve_state.reset_mock()
 
     await hass.services.async_call(

